packages/cli/src/commands/info/info_command.ts

```diff
@@ -0,0 +1,51 @@
+import * as os from 'node:os';
+import { Argv, CommandModule } from 'yargs';
+import { CdkInfoProvider } from '../../info/cdk_info.js';
+import { EnvironmentInfoProvider } from '../../info/env_info.js';
+import { printer } from '../../printer.js';
+
+/**
+ * Represents the InfoCommand class.
+ */
+export class InfoCommand implements CommandModule<object> {
+  /**
+   * @inheritDoc
+   */
+  readonly command: string;
+
+  /**
+   * @inheritDoc
+   */
+  readonly describe: string;
+
+  /**
+   * Creates top level entry point for generate command.
+   */
+  constructor(
+    private readonly environmentInfoProvider: EnvironmentInfoProvider,
+    private readonly cdkInfoProvider: CdkInfoProvider
+  ) {
+    this.command = 'info';
```

Comment:
just to play devil's advocate, is there anything else we may want to put behind an "info" command in the future? Maybe info about the currently deployed backend, or info about a local diff. I'm not opposed to `amplify info` just want to make sure we aren't backing ourselves in a corner. We should also make sure we have PM alignment on the name if we haven't already done that.

Reply:
`info` seems fine but to Ed's note I could see this growing or us wanting something else/more later. We've had thoughts of `diagnose`/`doctor`/something similar, and the motivation for `info` is to ease env info collection for filing issues. Maybe this is something that's rolled up in to a similar command?
```
amplify diagnose --env-info
```

Reply:
Sounds like `info` is probably good for now. We can add args / deprecate / shuffle things around in future major versions if necessary